lua/cfc_prop_restoration/core/sv/sv_init.lua

```diff
@@ -230,17 +220,45 @@ end
 
 hook.Add( "PlayerInitialSpawn", "CFC_Restoration_Reconnect", handleReconnect )
 
+local function saveProps( time )
+    if not table.IsEmpty( queue ) then return end
+
+    time = time or CurTime()
+
+    logger:debug( "Autosaving player props" )
+
+    local playersProps = getAllPlayerProps()
+
+    for _, ply in pairs( player.GetHumans() ) do
+        hook.Run( "CFC_PropRestore_SavingPlayer", ply )
+
+        local propVelocities = getEntityRestorers( playersProps[ply] )
+        local copyObj = PropRestoration.Copy( ply )
```

Comment:
`getAllPlayerProps` is meant to fix the problem of looping over all ents for every player.
It loops through all ents once and builds a table of props for each player. You can pass this table into `PropRestoration.Copy` instead of looping through all ents in that function also.

You could also change `getAllPlayerProps` to use the new `canDupe`, or you could just run `canDupe` on all of them again.

```suggestion
        local plyProps = playersProps[ply]
        local propVelocities = getEntityRestorers( plyProps )
        local copyObj = PropRestoration.Copy( plyProps )
```